vllm/entrypoints/openai/serving_chat.py

```diff
@@ -703,7 +703,6 @@ async def chat_completion_full_generator(
                             name=request.tool_choice.function.name,
                             arguments=output.text))
                     ])
-                tools_called = True
```

Comment:
Can you add some comments about the behavior we are going for here to match the OpenAI spec?
The whole name and use of this `tools_called` variable may need some refactoring.

Reply:
Yes, that makes sense